src/main/java/uk/gov/justice/laa/crime/cfecrime/utils/FullMeansTestOutcomeCalculator.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.laa.crime.cfecrime.utils;
+
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.justice.laa.crime.cfecrime.cma.enums.FullAssessmentResult;
+import uk.gov.justice.laa.crime.cfecrime.cma.enums.MagCourtOutcome;
+import uk.gov.justice.laa.crime.cfecrime.cma.enums.CaseType;
+import uk.gov.justice.laa.crime.cfecrime.cma.enums.MeansTestOutcome;
+import java.util.HashSet;
+import java.util.Set;
+
+@Slf4j
+public class FullMeansTestOutcomeCalculator {
+
+    public static MeansTestOutcome getFullMeansTestOutcome(FullAssessmentResult result, CaseType caseType, MagCourtOutcome magCourtOutcome){
+        log.debug("Get the outcome of the Full Means Test result = {} caseType = {} magCourtOutcome = {}", result, caseType, magCourtOutcome);
+
+        //Fail data
+        Set<CaseType> caseTypesHeardInMagistratesCourt = new HashSet<CaseType>();
+        caseTypesHeardInMagistratesCourt .add(CaseType.COMMITAL);
+        caseTypesHeardInMagistratesCourt .add(CaseType.SUMMARY_ONLY);
+        caseTypesHeardInMagistratesCourt .add(CaseType.EITHER_WAY);
+
+        MeansTestOutcome meansTestOutcome = MeansTestOutcome.ELIGIBLE_WITH_NO_CONTRIBUTION;
+        if (result != null && caseType != null ) {
+
+            if (result == FullAssessmentResult.PASS) {
```

Comment:
a switch would definitely read nicer here rather than 3 ifs IMHO

Reply:
I did have it as a switch, however as I said the code coverage does not like and reading about this it is a bug in Jacobo and they all suggest to get around this to use if-then-else.

Reply:
I would say that was tail wagging the dog. I would prefer cleaner easier to read code over some faulty code coverage metric

Reply:
I think we might need a new coverage tool